packages/react-query/src/types.ts

```diff
@@ -68,6 +78,11 @@ export type UseQueryResult<
   TError = unknown,
 > = UseBaseQueryResult<TData, TError>
 
+export type UseSuspenseQueryResult<TData = unknown, TError = unknown> = Omit<
+  DefinedQueryObserverResult<TData, TError>,
+  'isPlaceholderData' | 'promise'
```

Comment:
promise doesn’t exist in v4

```suggestion
  'isPlaceholderData' | 'isPreviousData'
```